pkg/controller/servicemesh/controlplane/readiness.go

```diff
@@ -100,12 +100,20 @@ func (r *controlPlaneInstanceReconciler) updateReadinessStatus(ctx context.Conte
 
 type isReadyFunc func(runtime.Object) bool
 
+// keep this in sync with kinds in calculateComponentReadiness()
+var kindsWithReadiness = sets.NewString("Deployment", "StatefulSet", "DaemonSet")
+
+func (r *controlPlaneInstanceReconciler) hasReadiness(kind string) bool {
+	return kindsWithReadiness.Has(kind)
+}
+
 func (r *controlPlaneInstanceReconciler) calculateComponentReadiness(ctx context.Context) (map[string]bool, error) {
 	readinessMap := map[string]bool{}
 	typesToCheck := []struct {
 		list  runtime.Object
 		ready isReadyFunc
 	}{
+		// keep this in sync with kindsWithReadiness
```

Comment:
Should we create a simple type for this?  Maybe make `kindsWithReadiness` a map, so that all of the related code is in one place?

Reply:
Yeah, I tried that, but found no simple solution, because we need a List-type in one place, and the base type in the other. And if I have to add type manipulation code, it's just going to be much more complicated that it is right now. On the other hand, this map and the set won't really change much (if ever), so I don't know if it makes sense to waste time improving this.